README.md

```diff
@@ -1,2 +1,117 @@
-# spire-mysql-demo
-Demonstration of using SPIRE-issued X.509-SVIDs to set up an authenticated mTLS connection to a MySQL database
+# MySQL mTLS Authentication using SPIRE
+
+Demonstration of using SPIRE-issued X.509-SVIDs in MySQL database and clients, providing client authentication
+and connection encryption in MySQL via mutual-TLS (mTLS), deployed on Kubernetes. This project assumes readers to have 
+basic knowledge of the following technologies and concepts:
+
+1. [SPIFFE](https://spiffe.io/) and [SPIRE](https://github.com/spiffe/spire)
+2. [Kubernetes](https://kubernetes.io/)
+3. [MySQL](https://www.mysql.com/)
+4. [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security)
+
+## [Getting Started](#getting-started)
+
+This section guides deploying and setting up components involved in this project on a Kubernetes cluster.
+Refer [design](#design) for details on the components involved.
+
+### Prerequisites
+
+1. `kubectl` is installed and available on the PATH: https://kubernetes.io/docs/tasks/tools/ 
+2. Kubernetes cluster is configured with `kubectl` and `kubectl context` is set to use this cluster
+3. `curl` is installed and available on the PATH: https://everything.curl.dev/get
+
+### Deploy and Setup
+
+Deploy SPIRE server and SPIRE agent in the `spire` namespace on the cluster.
+The script also creates necessary registrations in SPIRE server.
+```
+./01-deploy-spire.sh
+```
+
+Deploy MySQL server along with a persistent volume in the `mysql` namespace on the cluster
+```
+./02-deploy-mysql.sh
+```
+
+In a separate terminal, start port forwarding to MySQL server pod
+```
+kubectl -n mysql port-forward pod/mysql-0 3306:3306
```

Comment:
```suggestion
kubectl -n mysql port-forward service/mysql 3306:3306
```
1. On second look, probably cleaner to do based on the service rather than pod. I'll update the script comments as well to suggest that
2. I guess we should agree on what port to use locally 😄